fv3gfs/physics/stencils/physics.py

```diff
@@ -103,89 +220,102 @@ def setup_const_from_state(self, state: dict):
-        phy_state.qliquid = phy_state.qliquid * phy_state.delp
-        phy_state.qrain = phy_state.qrain * phy_state.delp
-        phy_state.qice = phy_state.qice * phy_state.delp
-        phy_state.qsnow = phy_state.qsnow * phy_state.delp
-        phy_state.qgraupel = phy_state.qgraupel * phy_state.delp
-        phy_state.qo3mr = phy_state.qo3mr * phy_state.delp
-        phy_state.qsgs_tke = phy_state.qsgs_tke * phy_state.delp
-        phy_state.delp = (
-            phy_state.delp
-            - phy_state.qliquid
-            - phy_state.qrain
-            - phy_state.qice
-            - phy_state.qsnow
-            - phy_state.qgraupel
-        )
-        self._prsi[:, :, 0] = self._ptop
-        for k in range(self.grid.npz):
-            self._prsi[:, :, k + 1] = self._prsi[:, :, k] + phy_state.delp[:, :, k]
-            self._prsik[:, :, k] = np.log(self._prsi[:, :, k])
-            phy_state.qvapor = phy_state.qvapor / phy_state.delp
-            phy_state.qliquid = phy_state.qliquid / phy_state.delp
-            phy_state.qrain = phy_state.qrain / phy_state.delp
-            phy_state.qice = phy_state.qice / phy_state.delp
-            phy_state.qsnow = phy_state.qsnow / phy_state.delp
-            phy_state.qgraupel = phy_state.qgraupel / phy_state.delp
-            phy_state.qo3mr = phy_state.qo3mr / phy_state.delp
-            phy_state.qsgs_tke = phy_state.qsgs_tke / phy_state.delp
-        self._prsik[:, :, -1] = np.log(self._prsi[:, :, -1])
-        self._prsik[:, :, 0] = np.log(self._ptop)
-        for k in range(self.grid.npz):
-            qgrs_rad = np.maximum(self._qmin, phy_state.qvapor[:, :, k])
-            rTv = rdgas * phy_state.pt[:, :, k] * (1.0 + con_fvirt * qgrs_rad)
-            self._dm[:, :] = phy_state.delp[:, :, k]
-            phy_state.delp[:, :, k] = (
-                self._dm * rTv / (self._phii[:, :, k] - self._phii[:, :, k + 1])
-            )
-            # if not hydrostatic, replaces it with hydrostatic pressure if violated
-            phy_state.delp[:, :, k] = np.minimum(
-                phy_state.delp[:, :, k], self._prsi[:, :, k + 1] - 0.01 * self._dm
-            )
-            phy_state.delp[:, :, k] = np.maximum(
-                phy_state.delp[:, :, k], self._prsi[:, :, k] + 0.01 * self._dm
-            )
-
-        self._prsik[:, :, -1] = np.exp(KAPPA * self._prsik[:, :, -1]) * self._pk0inv
-        self._prsik[:, :, 0] = self._pktop
-        return phy_state
-
     def __call__(self, state: dict, rank):
         self.setup_const_from_state(state)
-        phy = PhysicsState(state, self.grid)
-        physics_state = phy.physics_state
-        physics_state = self.prepare_physics_state(physics_state)
+        shape = self.grid.domain_shape_full(add=(1, 1, 1))
+        origin = self.grid.compute_origin()
+        storage = utils.make_storage_from_shape(shape, origin=origin, init=True)
+        state = get_namespace(DynamicalCore.arg_specs, state)
```

Comment:
We should refactor state at the top level to some MainLoopState, so this can take in `state: PhysicsState` and at the top level we can pass in `state.physics`. Requires some coordination with the dycore, so this is maybe easier to do in a mono-repo.

Reply:
This is a good point. I'm not sure how to consolidate this given `state` from dycore is a dictionary and arg get_namespace is called inside dycore (and replicated in physics). Do we want to make a state object for dycore as well?